payjoin/src/input_type.rs

```diff
@@ -1,14 +1,14 @@
 use std::convert::{TryFrom, TryInto};
 use std::fmt;
 
-use bitcoin::blockdata::script::{Instruction, Instructions, Script};
+use bitcoin::blockdata::script::{Instruction, Instructions, Script, ScriptBuf};
 use bitcoin::blockdata::transaction::TxOut;
-use bitcoin::util::psbt::Input as PsbtInput;
+use bitcoin::psbt::Input as PsbtInput;
 
 /// Takes the script out of script_sig assuming script_sig signs p2sh script
-fn unpack_p2sh(script_sig: &Script) -> Option<Script> {
+fn unpack_p2sh(script_sig: &Script) -> Option<ScriptBuf> {
     match script_sig.instructions().last()?.ok()? {
-        Instruction::PushBytes(bytes) => Some(Script::from(bytes.to_vec())),
+        Instruction::PushBytes(bytes) => Some(ScriptBuf::from(bytes.as_bytes().to_vec())),
```

Comment:
@kixunil Does this violate the new paradigm by calling to_vec and allocating? The Script / ScriptBuf / PushBytes / PushBytesBuf changes of the upgrade are the most confusing part me but I think I'm getting a handle on them.

Reply:
Yeah, the function should return `Option<&Script>` and this should be `Script::from_bytes` instead.

IMO this function should've been upstreamed anyway.